acceptance-tests/upgrade/update_and_upgrade_mssql_db_failover_group_subsume_test.go

```diff
@@ -0,0 +1,148 @@
+			}),
+		)
+		defer masbServiceInstance.Delete()
+
+		By("creating a failover group using the MASB broker")
+		fogName := random.Name(random.WithPrefix("fog"))
+		masbFOGServiceInstance := services.CreateInstance(
+			"azure-sqldb-failover-group",
+			"SecondaryDatabaseWithFailoverGroup",
+			services.WithMASBBroker(),
+			services.WithParameters(masbFOGConfig(masbDBName, fogName)),
+		)
+		defer masbFOGServiceInstance.Delete()
+
+		By("pushing the unstarted app")
+		app := apps.Push(apps.WithApp(apps.MSSQL))
+		defer apps.Delete(app)
+
+		By("binding the app to the MASB fog")
+		masbFOGServiceInstance.Bind(app)
+
+		By("starting the app")
+		apps.Start(app)
+
+		By("creating a schema using the app")
+		schema := random.Name(random.WithMaxLength(10))
+		app.PUT("", schema)
+
+		By("setting a key-value using the app")
+		key := random.Hexadecimal()
+		value := random.Hexadecimal()
+		app.PUT(value, "%s/%s", schema, key)
+
+		By("getting the value using the app")
+		got := app.GET("%s/%s", schema, key)
+		Expect(got).To(Equal(value))
+
+		By("Create the CSB with DB server details")
+		serverPairTag := random.Name(random.WithMaxLength(10))
+		serviceBroker := brokers.Create(
+			brokers.WithPrefix("csb-mssql-fog-db"),
+			brokers.WithSourceDir(releasedBuildDir),
+			brokers.WithEnv(apps.EnvVar{Name: "MSSQL_DB_FOG_SERVER_PAIR_CREDS", Value: serverPairsConfig(serverPairTag)}),
+		)
+		defer serviceBroker.Delete()
+
+		By("subsuming the database failover group")
+		dbFogInstance := services.CreateInstance(
+			"csb-azure-mssql-db-failover-group",
+			"subsume",
+			services.WithBroker(serviceBroker),
+			services.WithParameters(map[string]string{
+				"azure_primary_db_id":   fetchResourceID("db", masbDBName, metadata.PreProvisionedSQLServer),
+				"azure_secondary_db_id": fetchResourceID("db", masbDBName, metadata.PreProvisionedFOGServer),
+				"azure_fog_id":          fetchResourceID("failover-group", fogName, metadata.PreProvisionedSQLServer),
+				"server_pair":           serverPairTag,
+			}),
+		)
+		defer dbFogInstance.Delete()
+
+		By("purging the MASB FOG instance")
+		masbFOGServiceInstance.Purge()
+
+		By("updating to another plan")
+		dbFogInstance.Update("-p", "small")
+
+		By("pushing the development version of the broker")
+		serviceBroker.UpgradeBroker(developmentBuildDir)
+
+		By("upgrading service instance")
+		dbFogInstance.Upgrade()
+
+		By("updating the instance plan after upgrading the service instance")
+		dbFogInstance.Update("-p", "medium")
+
+		By("binding the app to the CSB service instance")
+		binding := dbFogInstance.Bind(app)
+		defer apps.Delete(app) // app needs to be deleted before service instance
+
+		By("restaging the app")
+		apps.Restage(app)
+
+		By("checking that the app environment has a credhub reference for credentials")
+		Expect(binding.Credential()).To(matchers.HaveCredHubRef)
+
+		By("getting the value set with the MASB binding")
+		got = app.GET("%s/%s", schema, key)
```

Comment:
Shouldn't we delete MASB binding or even the whole service instance before trying to run `app.GET("%s/%s", schema, key)`

How can we guarantee that the value we are fetching doesn't come from the still active MASB instance? We are not calling `Unbind` at any moment. And we aren't calling `masbServiceInstance.Delete()` either until the whole test has finished.

Reply:
I didn't see the instruction `masbFOGServiceInstance.Purge()`

I guess that's exactly what I was suggesting. Sorry.
My fault.

Reply:
After our discussion on Slack, it seems we still would like to add an Unbind operation.
However, as you mentioned, it is not mandatory because of the way the app was written (CSB bindings take precedence)